src/python/pants/rules/core/cloc.py

```diff
@@ -67,42 +67,43 @@ class CountLinesOfCode(Goal):
   console: Console,
   options: CountLinesOfCodeOptions,
   cloc_script: DownloadedClocScript,
-  hydrated_targets: HydratedTargets,
+  sources_snapshots: SourcesSnapshots,
 ) -> CountLinesOfCode:
   """Runs the cloc perl script in an isolated process"""
 
-  all_target_adaptors = {ht.adaptor for ht in hydrated_targets}
-  digests_to_merge = []
+  snapshots = [sources_snapshot.snapshot for sources_snapshot in sources_snapshots]
+  file_content = '\n'.join(
+    sorted(set(itertools.chain.from_iterable(snapshot.files for snapshot in snapshots)))
+  ).encode()
 
-  source_paths: Set[str] = set()
-  for t in all_target_adaptors:
-    sources = getattr(t, 'sources', None)
-    if sources is not None:
-      digests_to_merge.append(sources.snapshot.directory_digest)
-      for f in sources.snapshot.files:
-        source_paths.add(str(f))
-
-  file_content = '\n'.join(sorted(source_paths)).encode()
+  if not file_content:
+    return CountLinesOfCode(exit_code=0)
```

Comment:
This is a fix for running `./pants cloc2`. Originally, it failed with a very verbose message. Now, we gracefully exit with return code 0 to mirror the semantics of `./pants fmt2 lint2 test`.

I added a test for this.